src/System.Security.Principal.Windows/tests/SecurityIdentifierTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Security.Principal;
+using Xunit;
+
+public class SecurityIdentifierTests
+{
+    [Fact]
+    public void ValidateGetCurrentUser()
+    {
+        using (WindowsIdentity identity = WindowsIdentity.GetCurrent(false))
+        {
+            Assert.NotNull(identity.User);
+        }
+    }
+
+    [Fact]
+    public void ValidateToString()
+    {
+        string sddl = null;
+        using (WindowsIdentity me = WindowsIdentity.GetCurrent(false))
+        {
+            sddl = me.User.ToString();
+        }
+
+        Assert.NotNull(sddl);
+        Assert.NotEmpty(sddl);
+        Assert.StartsWith("S-1-5-", sddl); // sid prefix, version 1, user account type
+        Assert.NotEqual('-', sddl[sddl.Length - 1]);
+
+        string[] parts = sddl.Substring(6).Split('-');
+
+        Assert.NotNull(parts);
+        Assert.NotEmpty(parts);
+
+        Assert.All(parts, part => uint.TryParse(part, out uint _));
+    }
+
+    [Fact]
+    public void ValidateToStringUsingWhoami()
+    {
+        string librarySid = null;
+        using (WindowsIdentity me = WindowsIdentity.GetCurrent(false))
+        {
+            librarySid = me.User.ToString();
+        }
+
+        string windowsSid = null;
+        string processArguments = " /user";
+        // Call whois to get current sid
+        Process whoamiProcess = new Process();
+        whoamiProcess.StartInfo.FileName = "whoami.exe"; // whoami.exe is in system32, should already be on path
```

Comment:
Not on Windows Nano apparently
https://mc.dot.net/#/user/Wraith2/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/8e4cd636de18fe4b3ba7cee0bfc7b6d3afdbb208/workItem/System.Security.Principal.Windows.Tests/analysis/xunit/SecurityIdentifierTests~2FValidateToStringUsingWhoami

You could add `[ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsWindowsNanoServer))] ` to the test. It might need to be`[ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsWindowsNanoServer), nameof(PlatformDetection.IsWindowsServerCore)]` actually (neither of them probably have it)

Reply:
Instead of shelling out to whoami, can you just get hte value of `UserPrincipal.Current.Sid.Value` in the namespace `System.DirectoryServices.AccountManagement`? I just tried it and I get the right value, no parsing or exe's required.

Reply:
And it probably works on Nano

Reply:
I went with the simple exclusion on core and nano. The behaviour of the method won't be different on those platforms and all the main targets will give test coverage.

Reply:
The API call is simpler and quicker than spawning an exe. I would go that way unless I am missing something

Reply:
The [`UserPrincipal.Current`](https://github.com/dotnet/corefx/blob/ed05109d18912262db4d2a9f5bbc6327073beb79/src/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/User.cs#L213) implementation uses the change that i'm trying to test.

Reply:
Oh, fair enough. If this proves to be fragile, we will have to hand-write the interop calls.